### PR TITLE
SLA

### PR DESCRIPTION
**Pull request by osoukup**
This pull request introduces all the necessary functionality to be able to set tracker SLA start and end timestamps. This incorporates number of smaller changes and some big ones. Most importantly a new **sla** module is introduced which implements SLA definition parsing and the followup computation.

In the future the SLA definition should live somewhere outside the tooling and we should not even own it but for now I am just putting the YAML file in the SLA module. The fetching logic is not trivial and this is rather a PoC so it would make a little sense to start with the later steps before we have agreement on the initial ones.

Going commit-wise should hopefully give enough description itself but do not hesitate to ask.

Closes OSIDB-1428
Closes OSIDB-1393
Closes OSIDB-96

### CODE REVIEW

**Comment by osoukup** [comment]
@costaconrado I would be specifically glad for your review as your are the one familiar with **osim** module and there is a lot of in common. I also implemented some of the logic (equality check) at the same time as you and then during the rebase I used some parts from your version and some from mine. I guess that the most significant change (as it is user facing) is that I decided to use **is** and **is not** instead of (not) equals. It seems to me more human-language-like and the machine does not care for which word we used plus it aligns with a lot of properties starting with **is**. I am opened to discussion so please raise any concerns you have.

File: osidb/models.py [hunk 1]
```diff
@@ -1973,6 +1973,16 @@ def is_community(self) -> bool:
             name=self.ps_module, ps_product__business_unit="Community"
         ).exists()
 
+    @property
+    def is_compliance_priority(self) -> bool:
+        """
+        check and return whether this affect module and component combination is compliance priority
```
Comment (costaconrado) [inline osidb/models.py:1990]: I would recommend to mention this file without the url to it since it is only accessible internally.

**Review by costaconrado — APPROVED** [review]
LGTM. very nice structured PR, thanks for that.

**Review by Elkasitu — APPROVED** [review]

File: apps/osim/checks.py [hunk 2]
```diff
@@ -140,7 +148,13 @@ def desc2equals(self, check_desc):
...
 
                 # model fields with defined choices require uppercase letters
-                if getattr(getattr(getattr(self.model, attr), "field"), "choices"):
+                if (
+                    hasattr(getattr(self.model, attr), "field")
+                    and hasattr(getattr(getattr(self.model, attr), "field"), "choices")
+                    and getattr(getattr(getattr(self.model, attr), "field"), "choices")
+                    is not None
+                    or attr in self.TEXT_CHOICES_PROPERTIES
+                ):
```
[the remark this answers is not in the document]
Comment (osoukup) [reply apps/osim/checks.py:157]: Good point. I will have a look at it

**Pull request merged**